File: day01.py
from collections import Counter
import numpy as np

ids1, ids2 = np.loadtxt("inputs/day01.txt", dtype=int).T
print(np.sum(np.abs(np.sort(ids1) - np.sort(ids2))))

ids2_counts = Counter(ids2)
print(sum(id * ids2_counts[id] for id in ids1))

File: day01.jl
using DelimitedFiles

data = readdlm("inputs/day01.txt", Int64)
println(sum(abs.(sort(data[1:end, 1]) - sort(data[1:end, 2]))))

counts = zeros(Int64, maximum(data))
for el ∈ data[1:end, 2] counts[el] += 1
println(sum(data[1:end, 1] .* counts[data[1:end, 1]]))